airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -2228,6 +2228,14 @@ class Goals(CRMObjectIncrementalStream):
     scopes = {"crm.objects.goals.read"}
 
 
+class Leads(CRMSearchStream):
+    entity = "leads"
+    last_modified_field = "hs_lastmodifieddate"
+    associations = ["contacts", "companies"]
+    primary_key = "id"
+    scopes = {"crm.objects.contacts.read", "crm.objects.companies.read", "crm.objects.leads.read"}
```

Comment:
why do we need the contacts and companies read scopes?

it looks like only crm.objects.leads.read is required according to the documentation - https://developers.hubspot.com/docs/api/crm/leads

Reply:
In case of contacts and companies associations with the leads